### ISSUE TITLE
"x" format should be considered invariant for numeric types (MA0011)

### ISSUE THREAD

**Issue by omarjuul** [meziantou/Meziantou.Analyzer#744]
### Version of the Meziantou.Analyzer NuGet package

2.0.161

### Rule Identifier

MA0011

### Target Framework

net8.0-windows

### C# Language version

12.0

### Description

The Hexadecimal format specifier `x` for (integral) numeric types is [documented](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#hexadecimal-format-specifier-x) as invariant (not affected by NumberFormatInfo and thus not culture-sensitive).

I propose to no longer report MA0011 / MA0076 when the type is numeric and the format string is "x". A similar thing might apply for the [Binary format specifier `b`](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#binary-format-specifier-b), I haven't checked...

### Reproduction Steps

Minimal code:

```c#
using System;

var num = 33;
Console.WriteLine("Hex representation: " + num.ToString("x")); // MA0011 should not be reported
Console.WriteLine($"Hex representation: {num:x}..."); // MA0076 should not be reported

Console.WriteLine("Hex representation: " + num.ToString("N")); // MA0011 should be reported
Console.WriteLine($"Hex representation: {num:N}..."); // MA0076 should be reported
```

### Other information

Very similar to #495, ticket formatting and wording heavily inspired by that ticket.

**Comment by meziantou** [comment]
`x`, `X`, `X8` (any number) are now detected as culture invariant. I've also added the `B` format.
